OpenRA.Mods.Common/UpdateRules/Rules/20180923/ExtractHackyAIModules.cs

```diff
@@ -124,28 +210,66 @@ public override IEnumerable<string> UpdateActorNode(ModData modData, MiniYamlNod
 					}
 				}
 
-				if (supportPowerFields.Any(f => hackyAINode.ChildrenMatching(f).Any()))
+				if (!addCustomHarvModule)
 				{
-					var spNode = new MiniYamlNode("SupportPowerBotModule@" + aiType, "");
-					spNode.AddNode(requiresCondition);
-
-					foreach (var spf in supportPowerFields)
+					// We want the default harvester module to be enabled for every AI that didn't customise one of its fields,
+					// so we need to update RequiresCondition to be enabled on any of the conditions granted by these AIs,
+					// but only if the condition hasn't been added yet.
+					var requiresConditionNode = defaultHarvNode.LastChildMatching("RequiresCondition");
+					if (requiresConditionNode == null)
+						defaultHarvNode.AddNode(requiresCondition);
+					else
 					{
-						var fieldNode = hackyAINode.LastChildMatching(spf);
-						if (fieldNode != null)
-							fieldNode.MoveAndRenameNode(hackyAINode, spNode, "Decisions");
-					}
+						var oldValue = requiresConditionNode.NodeValue<string>();
+						if (oldValue.Contains(conditionString))
+							continue;
 
-					addNodes.Add(spNode);
+						requiresConditionNode.ReplaceValue(oldValue + " || " + conditionString);
```

Comment:
You are adding the same `requiresCondition` to all of the module type, and yaml nodes are a reference type so when you change this you actually change *all* nodes!

The fix is to create the `requiresCondition` fresh for each module you wish to add it to.